### DISCUSSION TITLE
Guides on using custom docker images and GPU

### DISCUSSION THREAD

**Discussion by bluematter** [fission/fission#2485] [Q&A] [answered]
I’m migrating our rendering engine in https://motionbox.io from lambda (which is cpu based) to having more control (gpu based functions)

I built a custom Docker image with nvidia/cuda/Ubuntu 

I have a node-pool in my GKE cluster that has GPUs and the drivers installed

My goal is to deploy an image to work with fission, nvidia, and node environment. Then each trigger boots up a new gpu function

is this possible? Maybe I’m missing docs, I’m also new to Kubernetes

**Comment by larivierec** [comment]
I know this is old but you should be able to use the runtimeClassName and the nodeSelector in the container pod spec

**Reply by sanketsudake** [reply]
@soharab-ic We should explore this further and share our findings.

**Reply by soharab-ic** [reply]
Sure @sanketsudake

**Reply by larivierec** [reply]
tl;dr; something like this

```yaml
---
apiVersion: node.k8s.io/v1
kind: RuntimeClass
metadata:
  name: nvidia
handler: "nvidia"
```

[using gpu example](https://github.com/larivierec/home-cluster/blob/main/kubernetes/main/apps/kube-system/nvidia/device-plugin/app/helm-release.yaml#L33-L42)

[example 2](https://github.com/larivierec/home-cluster/blob/main/kubernetes/main/apps/kube-system/nvidia/dcgm-exporter/app/helm-release.yaml#L34-L42)

**Comment by sanketsudake** [comment, marked as answer]
We have written a blog which briefly documents how to run GPU based functions.

https://fission.io/blog/running-gpu-based-functions-on-fission/
